### ISSUE TITLE
Novel view synthesis

### ISSUE THREAD

**Issue by mix345** [eriksandstroem/Point-SLAM#14]
First of all, thank you very much for sharing your fantastic work!

I have a question about novel view synthesis with the trained neural point clouds.

It appears that Point-SLAM uses gt-depth for view synthesis to sample evaluation points of volume rendering. 
(https://github.com/eriksandstroem/Point-SLAM/blob/main/src/Mapper.py#L844)

In situations where ground truth depth is not available for novel view synthesis, I set gt_depth=None in the renderer, but the generated image does not look very good with the default setting, displaying many voids.
(https://github.com/eriksandstroem/Point-SLAM/blob/main/src/utils/Renderer.py#L122)

Could you suggest an effective method for rendering images when ground truth depth is missing?

Thanks in advance!

**Comment by eriksandstroem** [comment]
Hi,
You are welcome! Happy that you appreciate it.

With regards to your question, yes, we did not focus on "novel view" rendering in our work, but only rendering from known poses. 

Currently, the way that we do rendering when no sensor depth is available is that we ray march and do more samples along the ray instead of jumping directly to the surface hypothesis given by the depth input. A more informed way to do novel view synthesis than what we have implemented so far would be to project the middle surface point for each ray (which we store in a separate variable) into the novel view. With some preprocessing to filter out occluded points, this should be possible to use as a replacement for the missing depth.

Hope this helps,
Erik
